src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskCompleteByIdControllerTest.java

```diff
@@ -147,137 +115,6 @@ public void should_return_a_403_if_task_was_not_previously_assigned() {
 
     }
 
-    @Test
-    public void should_return_a_204_when_completing_a_task_by_id_with_restricted_role_assignment() {
-
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "processApplication");
-        taskId = taskVariables.getTaskId();
-        initiateTask(taskVariables, Jurisdiction.WA);
-
-        common.setupRestrictedRoleAssignmentForWA(taskVariables.getCaseId(), caseworkerCredentials.getHeaders());
-        given.iClaimATaskWithIdAndAuthorization(
-            taskId,
-            caseworkerCredentials.getHeaders(),
-            HttpStatus.NO_CONTENT
-        );
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            taskId,
-            caseworkerCredentials.getHeaders()
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.NO_CONTENT.value());
-
-        assertions.taskVariableWasUpdated(taskVariables.getProcessInstanceId(), "taskState", "completed");
-        assertions.taskStateWasUpdatedInDatabase(taskId, "completed", caseworkerCredentials.getHeaders());
-
-        common.cleanUpTask(taskId);
-
-    }
-
-    @Test
-    public void should_return_a_401_when_the_user_did_not_have_any_roles() {
```

Comment:
This is a valid scenario where assign_and_complete is false? Should reintroduce or can move to IT

Reply:
This is already done in 1878 pr